index.d.ts

```diff
@@ -1,78 +1,150 @@
-
-  export class GoogleSigninButton extends React.Component<GoogleSigninButtonProps>{
-    Size: {
-      Icon: number;
-      Standard: number;
-      Wide: number;
-    };
-
-    Color: {
-      Auto: string;
-      Light: string;
-      Dark: string;
-    };
-  }
+// Definitions by: Jacob Froman <https://github.com/j-fro>
+//                 Michele Bombardi <https://github.com/bm-software>
+//                 Christian Chown <https://github.com/christianchown>
+//                 Eric Chen <https://github.com/echentw>
 
-  export type ConfigureParams = {
-    iosClientId?: string,
-    offlineAccess?: boolean,
-    webClientId?: string,
-    scopes?: string[],
-    hostedDomain?: string,
-    forceConsentPrompt?: boolean,
-    accountName?: string
-  };
+import * as React from 'react';
+import { ViewProps } from 'react-native';
 
-  export type HasPlayServicesParams = {
-    showPlayServicesUpdateDialog: boolean
-  };
+export interface GoogleSigninButtonProps extends ViewProps {
+  size?: GoogleSigninButton.Size;
+  color?: GoogleSigninButton.Color;
+  disabled?: boolean;
+  onPress?(): void;
+}
 
-  export type User = {
-    idToken: string,
-    accessToken: string | null,
-    accessTokenExpirationDate: number | null, // DEPRECATED, on iOS it's a time interval since now in seconds, on Android it's always null
-    serverAuthCode: string,
-    scopes: string[], // on iOS this is empty array if no additional scopes are defined
-    user: {
-      email: string,
-      id: string,
-      givenName: string | null,
-      familyName: string | null,
-      photo: string | null, // url
-      name: string // full name
-    }
-  }
+export class GoogleSigninButton extends React.Component<GoogleSigninButtonProps> {
+  constructor(props: GoogleSigninButtonProps);
+}
 
-  // Android Status codes: https://developers.google.com/android/reference/com/google/android/gms/auth/api/signin/GoogleSignInStatusCodes
-  export interface StatusCodes {
-    SIGN_IN_CANCELLED: string,
-    IN_PROGRESS: string,
-    PLAY_SERVICES_NOT_AVAILABLE: string,
-    SIGN_IN_REQUIRED: string
+export namespace GoogleSigninButton {
+  enum Size {
+    Standard,
+    Wide,
+    Icon
   }
 
-  export interface GoogleSignin {
-    hasPlayServices(params?: HasPlayServicesParams): Promise<boolean>
-    configure(params?: ConfigureParams): void
-    signInSilently(): Promise<User>
-    signIn(): Promise<User>
-    signOut(): Promise<void>
-    revokeAccess(): Promise<void>
-    isSignedIn(): Promise<boolean>
+  enum Color {
+    Light,
+    Dark
   }
 }
+
+export interface HasPlayServicesParams {
+  /**
+   * When showPlayServicesUpdateDialog is true, the user will be prompted to
+   * install Play Services if on Android and they are not installed.
+   * Default is true
+   */
+  showPlayServicesUpdateDialog?: boolean;
+}
+
+export interface ConfigureParams {
+  /**
+   * The Google API scopes to request access to. Default is email and profile.
+   */
+  scopes?: string[];
```

Comment:
this reminds me that `loginHint` is missing here and also in flow typings :)

Reply:
great catch!